From 67793af161b4b8fe474921a9e7e5041bfa6ec226 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Tue, 13 Nov 2012 13:55:45 +0100
Subject: [PATCH 058/793] net: xilinx_emacps: Simplify driver initialization

Use module_platform_driver macro which creates
init and exit functions.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Acked-by: John Linn <john.linn@xilinx.com>
---
 drivers/net/ethernet/xilinx/xilinx_emacps.c |   29 +--------------------------
 1 files changed, 1 insertions(+), 28 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_emacps.c b/drivers/net/ethernet/xilinx/xilinx_emacps.c
index 63e2362..a375655 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emacps.c
+++ b/drivers/net/ethernet/xilinx/xilinx_emacps.c
@@ -3275,34 +3275,7 @@ static struct platform_driver xemacps_driver = {
 	},
 };
 
-/**
- * xemacps_init - Initial driver registration call
- *
- * Returns whether the driver registration was successful or not.
- */
-static int __init xemacps_init(void)
-{
-    /*
-     * No kernel boot options used,
-     * so we just need to register the driver.
-     * If we are sure the device is non-hotpluggable, call
-     * platform_driver_probe(&xemacps_driver, xemacps_probe);
-     * to remove run-once probe from memory.
-     * Typical use for system-on-chip processor.
-     */
-	return platform_driver_probe(&xemacps_driver, xemacps_probe);
-}
-
-/**
- * xemacps_exit - Driver unregistration call
- **/
-static void __exit xemacps_exit(void)
-{
-	platform_driver_unregister(&xemacps_driver);
-}
-
-module_init(xemacps_init);
-module_exit(xemacps_exit);
+module_platform_driver(xemacps_driver);
 
 MODULE_AUTHOR("Xilinx, Inc.");
 MODULE_DESCRIPTION("Xilinx Ethernet driver");
-- 
1.7.5.4

